src/ray/raylet/scheduling_queue.h

```diff
@@ -116,21 +116,35 @@ class SchedulingQueue {
   void MoveTasks(std::unordered_set<TaskID> tasks, TaskState src_state,
                  TaskState dst_state);
 
+  class TaskQueue {
+  public:
+    TaskQueue() {}
+    bool Append(const TaskID &task_id, const Task &task);
+    bool Remove(const TaskID &task_id);
```

Comment:
I would change the name to  `RemoveTask` and `RemoveTasks` to make the difference explicit. Also, do we need `RemoveTask` for a single task if it's just a special case of `RemoveTasks`?

Reply:
will rename to ```AppendTask()``` and ```RemoveTask()``` respectively.  For the two ```Remove``` functions, they are actually both removing a single task, it's just one of them will append the removed task to the passed in vector if removal succeeds.